community/kernel/src/test/java/org/neo4j/kernel/impl/index/schema/SpatialIndexCacheTest.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (c) 2002-2018 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.kernel.impl.index.schema;
+
+import org.junit.Test;
+
+import java.util.Random;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.neo4j.helpers.collection.Iterators;
+import org.neo4j.values.storable.CoordinateReferenceSystem;
+
+import static org.junit.Assert.assertNull;
+
+public class SpatialIndexCacheTest
+{
+    @Test
+    public void stressCache() throws Exception
```

Comment:
in any concurrency releated tests its a good practice to set some timeout to make sure this will not gonna stuck ever